tests/test_data_types.py

```diff
@@ -70,8 +65,12 @@ def test_no_data():
     data_type = data_type_factory("float32", no_data=math.nan)
     assert math.isnan(data_type.no_data)
 
-    with pytest.raises(ValueError):
-        data_type_factory("float32", no_data=1.1)
+    data_type = data_type_factory("float32", no_data=0.0)
+    assert data_type.no_data == 0.0
+
+    # FIXME: Surely this test is wrong?
```

Comment:
is this something you meant to look at now, or later?

Reply:
Later. It was an existing test that verified what I can only imagine is incorrect behavior, but I want to check with Thomas when he comes back. I used a FIXME because it grabs attention, but should probably have just made a note or a ticket about it elsewhere. I'll remove it and do that.

Reply:
Before I was only accepting `None` and `math.isnan` for float values. Hence `1.1` is not an acceptable number although it is of type float.

`elif ("float" in dtype or dtype in ["half", "single", "double"]) and (no_data is not None and not math.isnan(no_data)`

If there is a usecase where we need to set actual float values and not `NaN` as no data values for float dataset we can change the behavior. Otherwise I prefer to keep it as is, since `NaN` makes most sense for this data type.